### PR TITLE
Respository restructure

### LINKED ISSUE #231: [Quality] Structure DAGs based on general type
## Current Situation

We have all of our DAGs side-by-side in one directory. This works fine for Airflow, but it might make sense to separate them out by type (possibly the same types as #222).

## Benefit

For contributors who wish to contribute providers, all of the other ingestion machinery we have can make it confusing to know what/where to contribute. Sequestering providers into their own subdirectory (along with all other DAG types) can provide some healthy compartmentalization.

## Additional context

Airflow has no trouble traversing into subdirectories. We'll need to be sure that we're ignoring non-DAG files appropriately though.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #262: Move one-off or obsolete DAGs to "retired" folder
## Current Situation

We have a number of DAGs (namely the `*_update_workflow.py` ones) that appear to be one-offs that no longer need to exist with the rest of the codebase.

## Suggested Improvement

These DAGs should be moved to a `retired` folder that is ignored by Airflow and separated from the rest of the DAGs.

## Benefit

This would reduce confusion for those who are perusing the codebase or attempting to find which DAGs are considered currently relevant.

## Additional context

Related to #231

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #231 by @AetherUnbound, fixes #262 by @AetherUnbound

## Description

There's no avoiding it, this is a **big PR**. A large number of files are modified, but most of those files have only been moved & had import paths changed. I'll enumerate all of the changes below, and try to include specific files that should be looked at that have more changes beyond moving/imports.

### Directory restructuring

I've moved files around in order to consolidate modules and make it easier to find things. The tests were also moved to mirror this new structure.

- Moved all `util/*` modules under `common/`
- Moved all `storage/*` modules under `common/`
- Moved provider API scripts and provider DAGs under `providers/`
- Moved all commoncrawl related code to `commoncrawl/`
- Moved other DAGs into subfolders related to function (`database/`, `maintenance/`, I didn't end up moving `common_api_workflows.py` because I wasn't sure where it should go!)

It was unclear to me why we had `common`, `util`, and `storage` as top level libraries, so I consolidated them all into `common`. If there was a rationale that I missed which makes separating these useful, let me know!

### Code changes

While doing this restructuring, I discovered that some tests had a hidden dependency; if I ran them in a different order, and unpredictable amount of tests would fail where they previously succeeded. After some tedious trial and error, I was able to track this down to **unexpected socket access**. The tests were set up with `pytest-socket` so that only tests which had explicitly enabled socket access could make `request` calls. Unfortunately, all tests after the `test_smithsonian_unit_codes.py` test circumvented this because of the pytest marker on the unit codes test! Adding socket exceptions for all of the failing tests seemed cumbersome and unhelpful, so I've removed that plugin entirely. If we want to assert that a test _shouldn't_ be accessing sockets, then we can add specific checks for said test. I don't think it's beneficial for us to block all requests across the board (at least at this time), seeing as how many requests are made for licenses alone. Instead, I've added the [`pytest-random-order` plugin](https://pypi.org/project/pytest-random-order/) and set this as the default run parameter for our CI. This will shuffle the tests so that they're run in a random order on GitHub. The plugin will report the seed it used, so we can use said seed locally if we ever encounter an issue with order-specific failures. I didn't set up a permanent configuration setting for this because we'd either have to 1) mount a `pytest.ini` file at the top level within the container along with the tests (it can't be in the test folder unless we want to run `pytest tests/` explicitly) or 2) add a new environment variable. The latter is plausible but this seemed to be the best balance between consistent behavior during development while still randomizing the checks during CI.

**TL;DR**: Tests had a hidden, order-specific requirement. The plugin that caused this has been removed and now tests are run in a random order on CI.

Other changes:

- Updated the `.airflowignore` file to respect the new structure
- Modified some recipes in the `justfile` to make multiple argument easier and run commands without having to start the `webserver` service
- Added directory structure documentation

---

I think that's it...thank you to whomever has the infinite patience necessary for making it this far :sweat_smile: :raised_hands:

## Testing Instructions

1. `just build`
1. `just test`
1. (Optional) `just run airflow dags list` to ensure all the relevant DAGs still get parsed

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
Madison, I've spent a lot of time looking for feedback to give you. With a PR of this size I hoped to have more specific feedback, but I'm really happy with the structure here and everything is working perfectly. Hopefully whoever else reviews sees things that I haven't, but great, great work here.

File: openverse_catalog/dags/common/loader/paths.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 from typing import Optional
 
 from airflow.models import TaskInstance
-from util.loader.ingestion_column import _fix_ingestion_column
+from common.loader.ingestion_column import _fix_ingestion_column
```
Comment (obulat) [inline openverse_catalog/dags/common/loader/paths.py:8]: Not related to this PR: I don't know if you've already noticed, but the `_fix_ingestion_column` can be retired :) It was used to convert old format into new, but we do not have that old format of tsvs anywhere anymore.

File: openverse_catalog/templates/create_api_script.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def fill_template(provider, media_type):
     script_template_path = TEMPLATES_PATH / "template_test.py_template"
     tests_path = REPO_PATH / "tests"
     # Mirror the directory structure, but under the "tests" top level directory
-    test_script_path = tests_path.joinpath(*api_path.parts[-2:]) / f"test_{filename}.py"
+    test_script_path = tests_path.joinpath(*api_path.parts[-3:]) / f"test_{filename}.py"
```
Comment (obulat) [inline openverse_catalog/templates/create_api_script.py:77]: This was such an easy change to miss! :)

File: tests/dags/test_dag_parsing.py [hunk 3]
```diff
@@ -10,52 +10,56 @@
...
+    ]
 ]
 
 # Extra DAGs to test
 ADDITIONAL_DAGS = [
-    "recreate_image_popularity_calculation.py",
-    "refresh_all_image_popularity_data.py",
-    "refresh_image_view_data.py",
-    "commoncrawl_etl.py",
+    "maintenance/airflow_log_cleanup_workflow.py",
```
Comment (obulat) [inline tests/dags/test_dag_parsing.py:40]: This is now so much easier to scan :)

**Review by obulat — APPROVED** [review]
These and previous changes make the repository structure much easier to understand.

The previous folder structure where `storage` was at the top was due to me having some problems with imports in the beginning, and not having enough experience with Airflow, so it is great to change it back :)

File: openverse_catalog/dags/common/loader/paths.py (hunk 1 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/common/loader/paths.py:8]: Great to know! I'll make a new issue for that 😄 I love that you keep pointing these out, I'm happy to take stuff out we no longer need to carry forward 💪
